### DISCUSSION TITLE
Improve validation for nested filter usage

### DISCUSSION THREAD

**Discussion by marianhlavac** [directus/directus#14544] [Feature Requests]
### Describe the Bug

Using nested filter with `_eq` results in HTTP 500 error on our Directus Cloud testing instance.

If the `_eq` is wrongly used, I'd expect an HTTP 400 error.

### To Reproduce

* Make a `GET` request to our instance at https://rnisc8xz.directus.app/items/news_post?filter[translations.languages_code][_eq]=en-US

### Errors Shown

```
{"errors":[{"message":"An unexpected error occurred.","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}
```

### What version of Directus are you using?

9.12.2

### What version of Node.js are you using?

Directus Cloud

### What database are you using?

Directus Cloud

### What browser are you using?

Safari

### How are you deploying Directus?

Directus Cloud

**Comment by Rigo-m** [comment]
Maybe related: https://github.com/directus/directus/issues/14276

**Comment by azrikahar** [comment]
We should use either `filter[translations][language_code][_eq]` or the `deep` operator when nesting[^1], as `filter[translations.language_code]` is not a valid syntax (but understandably it should error out more gracefully as pointed out). 

This is also mentioned in the [filter query](https://docs.directus.io/reference/query/#filter) in the docs, particularly the notice:

> **Nested Filters**
>
>The above example will filter the top level items based on a condition in the related item. If you're looking to filter the related items themselves, take a look at [the deep parameter](https://docs.directus.io/reference/query/#deep)!

[^1]: Note that the result will differ between them! Do refer to this elaborate comment that explains more on it: https://github.com/directus/directus/discussions/13459#discussioncomment-2806776

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
